testcase/integration/test_integration.py

```diff
@@ -0,0 +1,278 @@
+    Loopchain.end_line = 80 * peer_count * channel_count
+
+    # Generate configure files. Run only one time at the beginning of the test.
+    print("*--- Generate peer configure path list...")
+    peer_conf_path_list, channel_manage_data_path = \
+        generate_peer_conf_path_list_extended(peer_count=peer_count, channel_list=channel_list)
+    print("> peer_conf_path: ", peer_conf_path_list)
+    print("> channel_manage_data_path: ", channel_manage_data_path)
+
+    genesis_data = conftest.get_genesis_data(conf_path_list=peer_conf_path_list)
+
+    # Run Each peer
+    for peer_order in range(peer_count):
+        peer_conf_path = peer_conf_path_list[peer_order]
+        Loopchain.args = ["loop", peer_type, "-d", "-o", peer_conf_path]
+        proc_name = f"peer{peer_order}"
+
+        print(f"==========PEER_{peer_order} READY TO START ==========")
+        xprocess.ensure(proc_name, Loopchain)
+
+        # Store process infomation for terminate processes at the end of the test
+        proc_info = xprocess.getinfo(proc_name)
+        proc_info_list.append(proc_info)
+
+    print(f"==========ALL GREEN ==========")
+    time.sleep(0.5 * peer_count * channel_count)
+
+    yield True
+
+    # Executed after this fixture's scope ends.
+    for proc_info in proc_info_list:
+        proc_info.terminate()
+
+    time.sleep(3)  # For additional tests, need a moment to cool down.
+
+
+class TestLoopchain:
+    sent_tx_data = {}  # Sent tx data. Needed to be compared whether it equals with the queried one.
+    tx_hash_by_channel = {}  # Tx hashes. It collects return values of 'send_transaction'.
+
+    # @pytest.mark.skip
+    def test_health_check_before_test(self, request):
+        """Health check before test starts
+
+        Test steps:
+            1. Get genesis local file which is created when the test starts.
+            2. Get rich account's balance (God is not used. Generally third one.)
+            3. Get balance from 'All channels of All peers'.
+            4. Compare original balance with queried one.
+
+        Assertion Tests:
+            1. Queried balance == Expected balance
+            2. All queried balance has same value
+        """
+        global genesis_data
+        print("Genesis data: ", genesis_data)
+
+        peer_count = int(request.config.getoption("--peer-count"))
+        channel_count = int(request.config.getoption("--channel-count"))
+
+        billionaire = genesis_data["transaction_data"]["accounts"][2]
+        target_account = billionaire["address"]
+        expected_balance = int(billionaire["balance"], 16)
+        print("Address of billionaire account: ", target_account)
+
+        querried_balance_list = []
+
+        for peer_order in range(peer_count):
+            port = 9000 + (peer_order * 100)
+
+            for channel_num in range(channel_count):
+                url = f"http://localhost:{port}/api/v3/channel_{channel_num}"
```

Comment:
You can replace it with the method in utils. (Not only this, but also all parts of making url manually.)

```py
channel_name = f"channel_{channel_num}"
url = utils.normalize_request_url(port, conf.ApiVersion.v3, channel_name)
```

Reply:
`normalized_request_url` is useful. Thank you.